addons/yatl/event/event.gd

```diff
@@ -43,3 +44,190 @@ const StreamOfflineEvent: Resource = preload("./types/events.gd").StreamOfflineE
 const UserAuthorizationGrantEvent: Resource = preload("./types/events.gd").UserAuthorizationGrantEvent
 const UserAuthorizationRevokeEvent: Resource = preload("./types/events.gd").UserAuthorizationRevokeEvent
 const UserUpdateEvent: Resource = preload("./types/events.gd").UserUpdateEvent
+
+
+# Private constants
+
+const __EVENT_WSS_URL: String = "ws://127.0.0.1:25708"
+
+
+# Private variables
+
+var __pal: __PAL
+
+var __broardcaster_user_id: String
+var __client_id: String
+var __access_token: String
+
+var __connection: __WebSocketConnection
+
+var __websocket_id: String
+var __websocket_frequency_min: float
+
+var __time_since_last_message: float = 0.0
+
+var __connections: Dictionary = {
+	# key: String - event
+	# value: [[FuncRef, Array - binds], ...]
+}
+
+
+# Lifecycle methods
+
+func _init(
+	_pal: __PAL,
+	_broadcaster_user_id: String,
+	_client_id: String,
+	_access_token: String
+) -> void:
+	__pal = _pal
+	__broardcaster_user_id = _broadcaster_user_id
+	__client_id = _client_id
+	__access_token = _access_token
+
+
+func _process(delta: float) -> void:
+	if !__connection:
+		return
+
+	__time_since_last_message += delta
+
+
+# Pubic methods
+
+func connect_event(
+	_event: String,
+	_target: Object,
+	_method: String,
+	_binds: Array = []
+) -> int:
+	# TODO: Check if event is a valid event
+	# return 1
+
+	if !__connection && !yield(__establish_connection(__EVENT_WSS_URL), "completed"):
+		return 2 # Some kind of error
```

Comment:
This and other return numbers should probably use the [`@GlobalScope`](https://docs.godotengine.org/en/stable/classes/class_%40globalscope.html) Error enum

Reply:
I am going to have a look through and see if there is a good match for them, otherwise I might make new errors for this. The list looks pretty comprehensive, so I should be able to avoid needing to create a specific one here.